frontend/src/__tests__/cypress/cypress/tests/mocked/projects/modelServingNim.cy.ts

```diff
@@ -0,0 +1,341 @@
+
+      projectDetails.visit('test-project');
+
+      // Card is visible
+      projectDetailsOverviewTab
+        .findDeployedModelServingRuntime('Test Name')
+        .should('have.text', 'NVIDIA NIM');
+    });
+
+    it('should be blocked if failed to fetch NIM model list', () => {
+      initInterceptsToEnableNim({});
+      projectDetailsOverviewTab.visit('test-project');
+      cy.findByTestId('model-serving-platform-button').click();
+      nimDeployModal.shouldDisplayError(
+        'There was a problem fetching the NIM models. Please try again later.',
+      );
+      nimDeployModal.findSubmitButton().should('be.disabled');
+    });
+  });
+
+  describe('Enabling NIM', () => {
+    describe('When NIM feature is enabled', () => {
+      it("should allow deploying NIM from a Project's Overview tab when the only platform", () => {
+        initInterceptsToEnableNim({});
+        projectDetailsOverviewTab.visit('test-project');
+        cy.findByTestId('model-serving-platform-button').click();
+        nimDeployModal.shouldBeOpen();
+      });
+
+      it("should allow deploying NIM from a Project's Overview tab when multiple platforms exist", () => {
+        initInterceptorsValidatingNimEnablement({
+          disableKServe: false,
+          disableModelMesh: false,
+          disableNIMModelServing: false,
+        });
+        projectDetailsOverviewTab.visit('test-project');
+        projectDetailsOverviewTab
+          .findModelServingPlatform('nvidia-nim')
+          .findByTestId('model-serving-platform-button')
+          .click();
+        nimDeployModal.shouldBeOpen();
+      });
+
+      it("should allow deploying NIM from a Project's Models tab when the only platform", () => {
+        initInterceptsToEnableNim({});
+        projectDetails.visitSection('test-project', 'model-server');
+        cy.get('button[data-testid=deploy-button]').click();
+        nimDeployModal.shouldBeOpen();
+      });
+
+      it("should allow deploying NIM from a Project's Models tab when multiple platforms exist", () => {
+        initInterceptorsValidatingNimEnablement({
+          disableKServe: false,
+          disableModelMesh: false,
+          disableNIMModelServing: false,
+        });
+        projectDetails.visitSection('test-project', 'model-server');
+        projectDetails
+          .findModelServingPlatform('nvidia-nim-model')
+          .findByTestId('nim-serving-deploy-button')
+          .click();
+        nimDeployModal.shouldBeOpen();
+      });
+    });
+
+    describe('When NIM feature is disabled', () => {
+      it("should NOT allow deploying NIM from a Project's Overview tab when the only platform", () => {
+        initInterceptorsValidatingNimEnablement({
+          disableKServe: true,
+          disableModelMesh: true,
+          disableNIMModelServing: true,
+        });
+        projectDetailsOverviewTab.visit('test-project');
+        cy.findByTestId('model-serving-platform-button').should('not.exist');
```

Comment:
The initial part should be in a page object -- you should be doing the `.should` off page object methods.

Reply:
Yeah, the reason I did that here, in the comment below (and in a bunch of other places in this file 😃 ), is because both deploy buttons for _single-serving_ and _nim-serving_ are named the same; in our test cases,, only the nim-platform-card is available, so this works. We can add a "findNimServingDeployButton" and a "findSingleServingDeployButton", but it won't be pretty. Unless we can add a testid to the buttons...

![image](https://github.com/user-attachments/assets/b90b2841-55e1-456b-8837-669ade7d1da7)

Reply:
The parent => parent div has `nvidia-nim-platform-card`